### PR TITLE
shortcut Iterable.containsExactlyElementsOf implemented

### PR DESCRIPTION
**Pull request by tkech17**
______________________________________
I confirm that I have read the [Contributor Agreements v1.0](https://github.com/robstoll/atrium/blob/master/.github/Contributor%20Agreements%20v1.0.txt), agree to be bound on them and confirm that my contribution is compliant.

### CODE REVIEW

File: apis/infix-en_GB/atrium-api-infix-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/IterableContainsInOrderOnlyElementsOfAssertionSpec.kt [hunk 1]
```diff
@@ -15,6 +18,14 @@ class IterableContainsInOrderOnlyElementsOfAssertionSpec : Spek({
             }.toThrow<IllegalArgumentException>()
         }
     }
+
+    describe("contains o inGiven order and only elementsOf") {
```
Comment (robstoll) [inline apis/infix-en_GB/atrium-api-infix-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/IterableContainsInOrderOnlyElementsOfAssertionSpec.kt:22]: please change to 
```suggestion
    describe("containsExactlyElementsOf") {
```

File: apis/infix-en_GB/atrium-api-infix-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/IterableContainsInOrderOnlyElementsOfAssertionSpec.kt [hunk 2]
```diff
@@ -15,6 +18,14 @@ class IterableContainsInOrderOnlyElementsOfAssertionSpec : Spek({
...
+        it("passing an empty iterable throws an IllegalArgumentException") {
+            expect {
+                expect(listOf(1, 2)) containsExactlyElementsOf (listOf())
```
Comment (robstoll) [inline apis/infix-en_GB/atrium-api-infix-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/IterableContainsInOrderOnlyElementsOfAssertionSpec.kt:25]: change to
```suggestion
                expect(listOf(1, 2)) containsExactlyElementsOf listOf()
```

File: apis/infix-en_GB/atrium-api-infix-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/IterableContainsInOrderOnlyElementsOfAssertionSpec.kt [hunk 3]
```diff
@@ -41,6 +59,29 @@ class IterableContainsInOrderOnlyElementsOfAssertionSpec : Spek({
...
+        private val containsExactlyElementsOfShortcutFun: KFunction2<Expect<Iterable<Double>>, Iterable<Double>, Expect<Iterable<Double>>> =
+            Expect<Iterable<Double>>::containsExactlyElementsOf
+
+        private fun getContainsShortcutPair() = containsExactlyElementsOfShortcutFun.name to Companion::containsExactlyElementsOfShortcut
+
+        private fun containsExactlyElementsOfShortcut(
+            expect: Expect<Iterable<Double>>,
+            a: Double,
+            aX: Array<out Double>
+        ): Expect<Iterable<Double>> = expect containsExactlyElementsOf(a glue aX)
```
Comment (robstoll) [inline apis/infix-en_GB/atrium-api-infix-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/IterableContainsInOrderOnlyElementsOfAssertionSpec.kt:72]: please change to 
```suggestion
        ): Expect<Iterable<Double>> = expect containsExactlyElementsOf listOf(a, *aX)
```

File: apis/infix-en_GB/atrium-api-infix-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/IterableContainsInOrderOnlyElementsOfAssertionSpec.kt [hunk 4]
```diff
@@ -41,6 +59,29 @@ class IterableContainsInOrderOnlyElementsOfAssertionSpec : Spek({
...
+        private val containsExactlyElementsOfNullableShortcutFun: KFunction2<Expect<Iterable<Double?>>, Iterable<Double?>, Expect<Iterable<Double?>>> =
+            Expect<Iterable<Double?>>::containsExactlyElementsOf
+
+        private fun getContainsNullableShortcutPair() = containsExactlyElementsOfNullableShortcutFun.name to Companion::containsExactlyElementsOfNullableShortcut;
+
+        private fun containsExactlyElementsOfNullableShortcut(
+            expect: Expect<Iterable<Double?>>,
+            a: Double?,
+            aX: Array<out Double?>
+        ): Expect<Iterable<Double?>> = expect containsExactlyElementsOf(a glue aX)
```
Comment (robstoll) [inline apis/infix-en_GB/atrium-api-infix-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/IterableContainsInOrderOnlyElementsOfAssertionSpec.kt:83]: please change to (also for IterableContainsInAnyOrderAtLeast1ElementsOfAssertionsSpec, I missed it in the review)
```suggestion
        ): Expect<Iterable<Double?>> = expect containsExactlyElementsOf listOf(a, *aX)
```

Comment (tkech17) [reply apis/infix-en_GB/atrium-api-infix-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/IterableContainsInOrderOnlyElementsOfAssertionSpec.kt:83]: ok then i'll do it in same commit if it's not problem

Comment (robstoll) [reply apis/infix-en_GB/atrium-api-infix-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/IterableContainsInOrderOnlyElementsOfAssertionSpec.kt]: Can you please fix it also for IterableContainsInAnyOrderAtLeast1ElementsOfAssertionsSpec

File: apis/infix-en_GB/atrium-api-infix-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/IterableContainsInOrderOnlyElementsOfAssertionSpec.kt (hunk 3 above)
Comment (robstoll) [reply apis/infix-en_GB/atrium-api-infix-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/IterableContainsInOrderOnlyElementsOfAssertionSpec.kt]: can you please fix it also for IterableContainsInAnyOrderAtLeast1ElementsOfAssertionsSpec

File: apis/infix-en_GB/atrium-api-infix-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/IterableContainsInOrderOnlyElementsOfAssertionSpec.kt (hunk 4 above)
Comment (tkech17) [reply apis/infix-en_GB/atrium-api-infix-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/IterableContainsInOrderOnlyElementsOfAssertionSpec.kt]: i did it in other commit as in this commit i did not have that part of the code

Comment (robstoll) [reply apis/infix-en_GB/atrium-api-infix-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/IterableContainsInOrderOnlyElementsOfAssertionSpec.kt]: good as well

**Comment by robstoll** [comment]
@tkech17 thanks for your contribution :slightly_smiling_face: :+1:

**Pull request merged**